src/components/ha-selector/ha-selector-selector.ts

```diff
@@ -0,0 +1,286 @@
+            select: {
+              mode: "dropdown",
+              required: true,
+              options: Object.keys(SELECTOR_SCHEMAS)
+                .concat("manual")
+                .map((key) => ({
+                  label:
+                    localize(
+                      `ui.components.selectors.selector.types.${key}` as LocalizeKeys
+                    ) || key,
+                  value: key,
+                })),
+            },
+          },
+        },
+        ...(choice === "manual"
+          ? ([
+              {
+                name: "manual",
+                selector: { object: {} },
+              },
+            ] as const)
+          : []),
+        ...(SELECTOR_SCHEMAS[choice]
+          ? SELECTOR_SCHEMAS[choice].length > 1
+            ? [
+                {
+                  name: "",
+                  type: "expandable",
+                  title: localize("ui.components.selectors.selector.options"),
+                  schema: SELECTOR_SCHEMAS[choice],
+                },
+              ]
+            : SELECTOR_SCHEMAS[choice]
+          : []),
+      ] as const
+  );
+
+  protected render() {
+    let data;
+    let type;
+    if (this._yamlMode) {
+      type = "manual";
+      data = { type, manual: this.value };
+    } else {
+      type = Object.keys(this.value)[0];
+      const value0 = Object.values(this.value)[0];
+      data = {
+        type,
+        ...(typeof value0 === "object" ? value0 : []),
+      };
+    }
+
+    const schema = this._schema(type, this.hass.localize);
+
+    return html`<ha-card>
+      <div class="card-content">
+        <p>${this.label ? this.label : ""}</p>
+        <ha-form
+          .hass=${this.hass}
+          .data=${data}
+          .schema=${schema}
+          .computeLabel=${this._computeLabelCallback}
+          @value-changed=${this._valueChanged}
+        ></ha-form></div
+    ></ha-card>`;
+  }
+
+  private _valueChanged(ev: CustomEvent) {
+    ev.stopPropagation();
+    const value = ev.detail.value;
+
+    const type = value.type;
+    if (!type || typeof value !== "object" || Object.keys(value).length === 0) {
+      // not sure how this happens, but reject it
+      return;
+    }
+
+    const oldType = Object.keys(this.value)[0];
+    if (type === "manual" && !this._yamlMode) {
+      this._yamlMode = true;
+      this.requestUpdate();
+      return;
+    }
+    if (type === "manual" && value.manual === undefined) {
+      return;
+    }
+    if (type !== "manual") {
+      this._yamlMode = false;
+    }
+    delete value.type;
+
+    let newValue;
+    if (type === "manual") {
+      newValue = value.manual;
+    } else if (type === oldType) {
+      newValue = {
+        [type]: { ...(value.manual ? value.manual[oldType] : value) },
+      };
+    } else {
+      newValue = { [type]: { ...SELECTOR_DEFAULTS[type] } };
+    }
+
+    fireEvent(this, "value-changed", { value: newValue });
+  }
+
+  private _computeLabelCallback = (schema: any): string =>
```

Comment:
I tried to type this with the SchemaUnion the way other forms do, but it tells me:

```
error TS2589: Type instantiation is excessively deep and possibly infinite.

251     schema: SchemaUnion<ReturnType<typeof this._schema>>
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

So I'm not sure what that's about...